runtime/compiler/runtime/SignalHandler.c

```diff
@@ -1850,6 +1850,87 @@ UDATA jitAMD64Handler(J9VMThread* vmThread, U_32 sigType, void *sigInfo)
 	return J9PORT_SIG_EXCEPTION_CONTINUE_SEARCH;
 }
 
+#elif defined(TR_HOST_ARM64) && defined(TR_TARGET_ARM64)
+
+extern void jitHandleNullPointerExceptionTrap(void);
+extern void jitHandleInternalErrorTrap(void);
+
+UDATA jitARM64Handler(J9VMThread* vmThread, U_32 sigType, void* sigInfo)
+{
+	PORT_ACCESS_FROM_VMC(vmThread);
+
+	J9JITExceptionTable *exceptionTable = NULL;
+	J9JITConfig *jitConfig = vmThread->javaVM->jitConfig;
+
+	if (jitConfig) {
+		const char* infoName;
+		UDATA *pcPtr;
+		void *infoValue;
+		U_32 infoType;
+		J9JITExceptionTable *exceptionTable;
+
+		infoType = j9sig_info(sigInfo, J9PORT_SIG_CONTROL, J9PORT_SIG_CONTROL_PC, &infoName, &infoValue);
+		if (infoType != J9PORT_SIG_VALUE_ADDRESS) {
+			return J9PORT_SIG_EXCEPTION_CONTINUE_SEARCH;
+		}
+		pcPtr = (UDATA *) infoValue;
+
+		exceptionTable = jitConfig->jitGetExceptionTableFromPC(vmThread, *pcPtr);
+
+		if (!exceptionTable && J9PORT_SIG_FLAG_SIGBUS == sigType) {
+		   // We might be in a jit helper routine (like arraycopy) so look at the link register as well...
+		   UDATA *lrPtr;
+		   /* R30 is LR for aarch64 */
+		   infoType = j9sig_info(sigInfo, J9PORT_SIG_GPR, 30, &infoName, &infoValue);
+		   if (infoType != J9PORT_SIG_VALUE_ADDRESS) {
+		      return J9PORT_SIG_EXCEPTION_CONTINUE_SEARCH;
+		   }
+		   lrPtr = (UDATA *) infoValue;
+		   exceptionTable = jitConfig->jitGetExceptionTableFromPC(vmThread, *lrPtr);
+		   if (exceptionTable) {
+				vmThread->jitException = (J9Object *) (*lrPtr);  /* the lr points at the instruction after the helper call */
+				*pcPtr = (UDATA) ((void *) &jitHandleInternalErrorTrap);
+				return J9PORT_SIG_EXCEPTION_CONTINUE_EXECUTION;
+		   }
+		}
+
+		if (exceptionTable) {
+			switch (sigType) {
+			case J9PORT_SIG_FLAG_SIGBUS:
+			case J9PORT_SIG_FLAG_SIGSEGV:
+				infoType = j9sig_info(sigInfo, J9PORT_SIG_SIGNAL, J9PORT_SIG_SIGNAL_INACCESSIBLE_ADDRESS, &infoName, &infoValue);
+				if (sigType == J9PORT_SIG_FLAG_SIGSEGV && infoType == J9PORT_SIG_VALUE_ADDRESS) {
+					if ( *(UDATA*)infoValue > 0xFFFF ) {
```

Comment:
You may want to define a symbol for a page size.
Is the size of the first page 64KB?

Reply:
We assume that the page size is 4KB. Defined `NUMBEROFBYTESINACCESSIBLE` as 4096 and changed the code to use it.